org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/IXMLDocumentProvider.java

```diff
@@ -31,4 +34,8 @@
 	 *         null otherwise.
 	 */
 	DOMDocument getDocument(String uri);
+	
+	default Collection<String> allDocuments() {

```

Comment:
Why you need this new method?

Reply:
Once classpath of a project changes the schemas located inside jars can be affected and hence i need to re-validate all XML files from that project.
I have a project and its URI and now i need to find all affected documents to call `XMLTextDocumentService#triggerValidationFor(...)`.
I accomplished this by iterating over all documents known to Doc provider (which is really a wrapper around `XMLTextDocumentService`) and checking it belongs to a project, check if doc URI starts with project URI.

Reply:
Perhaps it should be better to have `Collection<DOMDocument> allDocuments()` ?